Source/WTF/wtf/generic/RunLoopGeneric.cpp

```diff
@@ -234,29 +279,27 @@ RunLoop::CycleResult RunLoop::cycle(RunLoopMode)
     return CycleResult::Continue;
 }
 
-void RunLoop::scheduleWithLock(Ref<TimerBase::ScheduledTask>&& task)
+bool RunLoop::isScheduledWithLock(const TimerBase::ScheduledTask& task) const
 {
-    m_schedules.append(task.ptr());
-    std::push_heap(m_schedules.begin(), m_schedules.end(), TimerBase::ScheduledTask::EarliestSchedule());
+    return m_schedules.findIf(TimerBase::ScheduledTask::IdenticalSchedule(task)) != notFound;
```

Comment:
This looks like `O(N)` operation. Can we keep `O(logN)`?
One possible implementation I can think is that, instead of using Vector for m_schedules, making it `RedBlackTree` (see `wtf/RedBlackTree.h`).
We can do that, so long as we modify `scheduledTimePoint` is not changed while it is in this tree. (And I think this is right. We are modifying it only after removing the node from this vector).

Reply:
Originally, when I created RunLoopGeneric, I cheated this since my design only removes a node when firing scheduled events. My design did not remove an arbitrary node from the vector, so I can just use `max-heap` (`std::push_heap` and `std::pop_heap`) here. But now, we need to have a way to remove arbitrary node efficiently, so I think we need balanced tree now :)